src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/Filters/DuplicatedDependenciesSnapshotFilter.cs

```diff
@@ -30,52 +30,59 @@ internal sealed class DuplicatedDependenciesSnapshotFilter : DependenciesSnapsho
             IAddDependencyContext context)
         {
             IDependency matchingDependency = null;
-            foreach ((string _, IDependency x) in context)
+            bool shouldApplyAlias = false;
+
+            foreach ((string _, IDependency other) in context)
             {
-                if (x.TopLevel 
-                     && !x.Id.Equals(dependency.Id, StringComparison.OrdinalIgnoreCase)
-                     && StringComparers.DependencyProviderTypes.Equals(x.ProviderType, dependency.ProviderType)
-                     && x.Caption.Equals(dependency.Caption, StringComparison.OrdinalIgnoreCase))
+                if (!other.TopLevel ||
+                    other.Id.Equals(dependency.Id, StringComparison.OrdinalIgnoreCase) ||
+                    !StringComparers.DependencyProviderTypes.Equals(other.ProviderType, dependency.ProviderType))
                 {
-                    matchingDependency = x;
-                    break;
+                    continue;
                 }
-            }
 
-            // If found node with same caption, or if there were nodes with same caption but with Alias already applied
-            // NOTE: Performance sensitive, so avoid formatting the Caption with parenthesis if it's possible to avoid it.
-            bool shouldApplyAlias = matchingDependency != null;
-            if (!shouldApplyAlias)
-            {
-                int adjustedLength = dependency.Caption.Length + " (".Length;
-                foreach ((string _, IDependency x) in context)
+                if (other.Caption.StartsWith(dependency.Caption, StringComparison.OrdinalIgnoreCase))
                 {
-                    if (x.TopLevel
-                         && !x.Id.Equals(dependency.Id, StringComparison.OrdinalIgnoreCase)
-                         && StringComparers.DependencyProviderTypes.Equals(x.ProviderType, dependency.ProviderType)
-                         && x.Caption.StartsWith(dependency.Caption, StringComparison.OrdinalIgnoreCase)
-                         && x.Caption.Length >= adjustedLength
-                         && string.Compare(x.Caption, adjustedLength, x.OriginalItemSpec, 0, x.OriginalItemSpec.Length, StringComparison.OrdinalIgnoreCase) == 0)
+                    if (other.Caption.Length == dependency.Caption.Length)
                     {
+                        // Exact match.
+                        System.Diagnostics.Debug.Assert(matchingDependency == null, "matchingDependency should be null");
+                        matchingDependency = other;
                         shouldApplyAlias = true;
                         break;
                     }
+
+                    // Prefix matches.
+                    // Check whether we have a match of form "Caption (ItemSpec)".
+
+                    string itemSpec = other.OriginalItemSpec;
+                    int expectedItemSpecIndex = dependency.Caption.Length + 2;
```

Comment:
I found the previous way this was adding this more obvious; can you put this `// Account for " ("` or somethign after it?

Reply:
Comment added.

It'd be nice if `" (".Length` was reduced to a constant by the compiler.